python/tskit/tables.py

```diff
@@ -24,55 +24,82 @@
 Tree sequence IO via the tables API.
 """
 import base64
-import collections
 import datetime
 import json
 import warnings
 
+import attr
 import numpy as np
 
 import _tskit
 import tskit
 import tskit.provenance as provenance
 import tskit.util as util
 
+attr_options = {"slots": True, "frozen": True, "auto_attribs": True}
 
-IndividualTableRow = collections.namedtuple(
-    "IndividualTableRow", ["flags", "location", "metadata"]
-)
 
+@attr.s(**attr_options)
```

Comment:
Jeez, this is witchcraft! We'll need to figure out which version of attrs we need for this all right though.